sdk/storage/src/blobs/append_blob_client.cpp

```diff
@@ -83,6 +83,13 @@ namespace Azure { namespace Storage { namespace Blobs {
     protocolLayerOptions.IfUnmodifiedSince = options.AccessConditions.IfUnmodifiedSince;
     protocolLayerOptions.IfMatch = options.AccessConditions.IfMatch;
     protocolLayerOptions.IfNoneMatch = options.AccessConditions.IfNoneMatch;
+    if (m_customerProvidedKey.HasValue())
+    {
+      protocolLayerOptions.EncryptionKey = m_customerProvidedKey.GetValue().Key;
+      protocolLayerOptions.EncryptionKeySha256 = m_customerProvidedKey.GetValue().KeyHash;
```

Comment:
Was the key hash created with hashing `EncryptionKey`? If so, I suppose we should not ask customer to input both right?

Reply:
I had the same thought as you. But current impl aligns with other SDK.